share/eds/byzantine/bad_encoding.go

```diff
@@ -104,7 +109,11 @@ func (p *BadEncodingProof) UnmarshalBinary(data []byte) error {
 // Validate checks that provided Merkle Proofs correspond to the shares,
 // rebuilds bad row or col from received shares, computes Merkle Root
 // and compares it with block's Merkle Root.
-func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
+func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
+	header, ok := hdr.(*header.ExtendedHeader)
+	if !ok {
+		return fmt.Errorf("invakud header type: expected %T, got %T", header, hdr)
```

Comment:
Wonderting if we should panic here instead. The binding between Proof and Header time is not a runtime concern and if the types do not do much something really wrong is going on.

Reply:
i'd be in favour of panicking